openmdao/core/system.py

```diff
@@ -3055,6 +3056,14 @@ def get_design_vars(self, recurse=True, get_sizes=True, use_prom_ivc=True):
                             else:
                                 out[name] = meta
 
+        if out:
+            for conn_in, conn_out in conns.items():
```

Comment:
Couldn't this check just be done once at the top level? 'conns' in this case is the top level connections dict, which can be quite large, and you're iterating over it at every level in the system tree.